src/app/feature/map/search/search-bar/search-bar.component.ts

```diff
@@ -1,75 +1,60 @@
-/* eslint-disable @typescript-eslint/naming-convention */
 import { Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
-import { AbstractControl, FormBuilder, FormGroup } from '@angular/forms';
+import { AbstractControl, FormBuilder } from '@angular/forms';
 import { Observable } from 'rxjs';
 import { map, startWith } from 'rxjs/operators';
+import { filterSearchValue } from 'src/app/core/utils';
+import { environment } from 'src/environments/environment';
 import { GuideMapFeaturePoint } from '../../../../core/models';
-
-// TODO: Moved interfaces to another place
-export interface IOption {
-  value: string;
-  viewValue: string;
-}
-
-export interface IOptionGroup {
-  groupName: string;
-  rooms: IOption[];
-}
-
-export const $filter = (opt: IOption[], value: string): IOption[] => {
-  const filterValue = value.toLowerCase();
-
-  return opt.filter(
-    (item) => item.viewValue.toLowerCase().indexOf(filterValue) === 0
-  );
-};
+import { IOptionGroup } from '../../interfaces';
 
 @Component({
   selector: 'app-search-bar',
   templateUrl: './search-bar.component.html',
   styleUrls: ['./search-bar.component.scss'],
 })
 export class SearchBarComponent implements OnInit {
-  constructor(private readonly fb: FormBuilder) {}
+  constructor(private readonly _formBuilder: FormBuilder) {}
 
   @Input() svgIconUrl: string;
```

Comment:
Please check it 
![image](https://user-images.githubusercontent.com/29804763/117782945-c6595780-b24a-11eb-84ed-a989dffbc3b4.png)